judgments/views/detail.py

```diff
@@ -79,23 +84,26 @@ def detail(request, document_uri):
     context = {}
 
     press_summary_suffix = "/press-summary/1"
-    if document_uri.endswith(press_summary_suffix):
-        context["document_type"] = "press_summary"
+    context["document_noun"] = document.document_noun
+    if document.best_human_identifier is None:
+        raise NoNeutralCitationError(document.uri)
+    context["judgment_ncn"] = document.best_human_identifier
+
+    if document.document_noun == "press summary":
         context["linked_document_uri"] = document_uri.removesuffix(press_summary_suffix)
-        context["judgment_ncn"] = ""
     else:
-        context["document_type"] = "judgment"
         context["linked_document_uri"] = document_uri + press_summary_suffix
-        context["judgment_ncn"] = document.neutral_citation
 
-    linked_document = None
     try:
         linked_document = get_published_document_by_uri(context["linked_document_uri"])
     except Http404:
         context["linked_document_uri"] = ""
-
-    if context["document_type"] == "press_summary" and linked_document:
-        context["judgment_ncn"] = linked_document.neutral_citation
+    else:
```

Comment:
I think we can skip the else to reduce the levels of nesting

the line `if context["document_noun"] == "press summary" and linked_document:` 

checks for the linked_document being non Falsey which can only be the case if the try block succeeded.